app/src/apps/osfi.py

```diff
@@ -45,24 +64,81 @@ def get_pie(data, column):
                 width=12,
             )
         ),
+        dbc.Row(
+            dbc.Col(
+                build_figure_container(
+                    title="Évolution temporelles des émissions (électricité)", id="electricity_time_series", footer="Explications..."
+                ),
+                width=12,
+            )
+        ),
+        dbc.Row(
+            dbc.Col(
+                build_figure_container(
+                    title="Évolution temporelles des émissions (gaz)", id="gaz_time_series", footer="Explications..."
+                ),
+                width=12,
+            )
+        ),
     ]
 )
 
 
 @app.callback(
     [
         Output("osfi-all-data-table", "columns"),
+        Output("osfi-all-data-table", "row_selectable"),
+        Output("osfi-all-data-table", "style_data"),
+        Output("osfi-all-data-table", "style_cell_conditional"),
         Output("osfi-all-data-table", "data"),
-        Output("emission-electricity-pie", "figure"),
-        Output("emission-gas-pie", "figure"),
     ],
     [Input("dashboard-selected-entity", "children")],
 )
+
 def update_graphs(selected_entity):
     service = oc.get_entity_by_id(selected_entity)
     data = oh.get_structure_data(service.code_osfi)
     columns = [{"name": i, "id": i} for i in data.columns]
+    columns = [{"name": i, "id": i} for i in data.loc[:,["Nom du bien"]]]
+    row_selectable = "multi"
+    selected_rows=[]
+    style_data={'whiteSpace': 'normal','height': 'auto','minWidth': '60px', 'width': '180px', 'maxWidth': '180px'}
+    style_cell_conditional =[{'if': {'column_id': i}, 'textAlign': 'left'} for i in ['Nom du bien','Date']]
     data_to_return = data.to_dict("records")
-    electricity_pie_graph = get_pie(data, "emission_electricity")
-    gas_pie_graph = get_pie(data, "emission_gaz")
-    return columns, data_to_return, electricity_pie_graph, gas_pie_graph
+    return columns, row_selectable, style_data, style_cell_conditional, data_to_return
+
+
+@app.callback(
+    [
+        Output("emission-electricity-pie", "figure"),
+        Output("emission-gas-pie", "figure"),
+        Output("electricity_time_series", "figure"),
+        Output("gaz_time_series", "figure"),
+    ],
+    [
+        Input("osfi-all-data-table", "derived_virtual_data"),
+        Input("osfi-all-data-table", "derived_virtual_selected_rows"),
+        Input("osfi-all-data-table", "selected_rows"),
+        Input("osfi-all-data-table", "data"),
+    ],
+    )
+def update_graphs_selected(derived_virtual_data, derived_virtual_selected_rows, selected_rows,data):
+    if derived_virtual_selected_rows is None:
+        derived_virtual_selected_rows = []
```

Comment:
You don't seem to be using the derived data anymore
```suggestion
    [
        Input("osfi-all-data-table", "selected_rows"),
        Input("osfi-all-data-table", "data"),
    ],
    )
def update_graphs_selected(selected_rows, data):
```